### PR TITLE
handle None outputs in snowflake IO manager

### PR DESCRIPTION
**Pull request by sryza**

### CODE REVIEW

File: python_modules/libraries/dagster-snowflake/dagster_snowflake/db_io_manager.py [hunk 1]
```diff
@@ -84,17 +84,27 @@ def __init__(self, type_handlers: Sequence[DbTypeHandler], db_client: DbClient):
...
+            )
+        else:
+            check.invariant(
+                context.dagster_type.is_nothing,
+                "Unexpected 'None' output value. If a 'None' value is intentional, set the output type to None.",
+            )
+            # if obj is None, assume that I/O was handled in the op body
+            handler_metadata = {}
 
         context.add_output_metadata(
```
Comment (jamiedemaria) [inline python_modules/libraries/dagster-snowflake/dagster_snowflake/db_io_manager.py:109]: i'm not 100% sure what this output metadata is used for but does it make sense to add the metadata about the query when the output is None?

**Review by jamiedemaria — APPROVED** [review]
looks good! one question about metadata, but otherwise 👍

**Comment by sryza** [comment]
@OwenKephart - thoughts on whether we should instead have some dbt-specific handling?

File: python_modules/libraries/dagster-snowflake/dagster_snowflake/db_io_manager.py (hunk 1 above)
Comment (sryza) [reply python_modules/libraries/dagster-snowflake/dagster_snowflake/db_io_manager.py:109]: This gives users a statement that they can copy/paste into the Snowflake interface to inspect the contents of their table.  How the table is created (by the I/O manager vs. in the op body) shouldn't affect the relevance of this query.

Comment (jamiedemaria) [reply python_modules/libraries/dagster-snowflake/dagster_snowflake/db_io_manager.py:109]: makes sense, thx for the clarification

**Comment by jamiedemaria** [comment]
I coincidentally just started working on dbt tutorial stuff and am using the snowflake io manager becuase i was able to get it set up the quickest for dev work. I pulled the changes from this PR into my branch because I immediately ran into issue with the dbt_run op returning None outputs. With the changes brought in, the `check.invariant` call that confirms that the type annotation is also None fails. I assume that means that the op we generate for dbt run needs to have `None` type annotations added to it (or the invariant check could accept Any as a valid type?)

**Comment by sryza** [comment]
@jamiedemaria ah yeah I made a separate PR for that: https://github.com/dagster-io/dagster/pull/9822

**Comment by OwenKephart** [comment]
What would dbt-specific handling look like for this?

**Comment by sryza** [comment]
> What would dbt-specific handling look like for this?

Maybe:
- The dbt op returns value of a `DbtOutput` type
- `build_snowflake_io_manager([

**Pull request merged**